lancero_source.go

```diff
@@ -236,6 +236,10 @@ func (ls *LanceroSource) ConfigureMixFraction(mfo *MixFractionObject) ([]float64
 
 // Sample determines key data facts by sampling some initial data.
 func (ls *LanceroSource) Sample() error {
+	// Cannot start this process if configuration failed.
+	if ls.configError != nil {
```

Comment:
This check should go before the first line of `Start` not in `Sample`, the first line of `Start` called `SetStateStarting` which probably changes internal state we'd rather not change.

Reply:
Cannot set it in `Start`, because that's a function using only the methods of the interface `DataSource`. We would need to add a method to `DataSource` to accomplish this.

Reply:
I suppose we could override `AnySource.SetStateStarting` method to a special version for `LanceroSource.SetStateStarting`. I'd rather keep it the way I had it at first: less confusing. But I do take your point about changing internal state.

Reply:
It's probably fine, the source isn't supposed to work anyway, maybe just add a comment about why it is done this way and how it "should" be done.

Reply:
Okay. See latest commit 2a59540. This is perhaps an improvement on the internal state in case errors are returned by `Sample` or `PrepareRun`.